utopia-remix/app/routes/v1.project.$id.tsx

```diff
@@ -20,8 +23,47 @@ export async function loader(args: LoaderFunctionArgs) {
 export async function action(args: ActionFunctionArgs) {
   return handle(args, {
     PUT: {
+      handler: createProject,
+      validator: ALLOW,
+    },
+    POST: {
       handler: proxy,
       validator: ALLOW,
     },
   })
 }
+
+async function createProject(req: Request, params: Params<string>) {
+  const { id } = params
+  ensure(id != null, 'project id is null', Status.BAD_REQUEST)
+
+  // create the project
+  const project = await proxy(req)
+
+  // handle access level setting
+  const url = new URL(req.url)
+  const accessLevelParam = url.searchParams.get('accessLevel')
+  const accessLevel: AccessLevel | null = accessLevelParamToAccessLevel(accessLevelParam)
+  if (accessLevel != null) {
+    await setProjectAccess({ projectId: id, accessLevel: accessLevel })
+  }
+
+  return project
+}
+
+function accessLevelParamToAccessLevel(accessLevelParam: string | null): AccessLevel | null {
```

Comment:
would it make sense to just pass the same integer values of the `AccessLevel` enum instead? So we'd have to do less parsing (and it can be put in the request body rather than the query params) 🤔

Reply:
the request body is the project data itself, and since the request is being passed "as-is" to the proxy (and I know there is a type validation on the project data there) I was not sure that I could add arbitrary fields directly in the body. If we can tweak it somehow then it's better of course.
As for the enum vs string - I thought it's better for the client to not know the internal enum representation of the access levels, but rather use a descriptive easy-to-read strings, do you think we need to opt for the enum values here?
btw in any case we would have to do some parsing in order to "cast" the number to the AccessLevel type (like `asAccessLevel` in `types.ts`)

Reply:
oooh you're right, the body is the project data! nvm, nvm. And the reasoning for the value makes sense, up to you :+1: